migration/v0.43/apply/10-kube-prometheus-stack.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+ROOT="$(readlink -f "$(dirname "${0}")/../../../")"
+
+source "${ROOT}/scripts/migration/lib.sh"
+
+run() {
+  case "${1:-}" in
+  execute)
+    chart_version=$(yq4 '.version' "${ROOT}/helmfile.d/upstream/prometheus-community/kube-prometheus-stack/Chart.yaml")
+    clusters=("${CK8S_CLUSTER}")
+    if [[ "${CK8S_CLUSTER}" == "both" ]]; then
+      clusters=("wc" "sc")
+    fi
+
+    for cluster in "${clusters[@]}"; do
+      current_version=$(helm_do "${cluster}" get metadata -n monitoring kube-prometheus-stack -ojson | jq -r '.version')
+
+      log_info "  - Checking if kube-promethes-stack needs to be upgraded"
+      if [[ "${current_version}" == "${chart_version}" ]]; then
+        log_info "  - ${cluster} already up to date, skipping"
+        continue
+      fi
+
+      log_info "  - Replace kube-prometheus-stack CRDs on ${cluster}"
+      kubectl_do "${cluster}" apply --server-side --force-conflicts -f "${ROOT}"/helmfile.d/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds
+
+      log_info "  - Upgrade kube-prometheus-stack on ${cluster}"
+      helmfile_upgrade "${cluster}" app=prometheus
+    done
```

Comment:
```suggestion
    for cluster in "${clusters[@]}"; do
      current_version=$(helm_do "${cluster}" get metadata -n monitoring kube-prometheus-stack -ojson | jq -r '.version')

      log_info "  - Checking if kube-promethes-stack CRDs needs to be upgraded"
      if [[ "${current_version}" != "${chart_version}" ]]; then
        log_info "  - Replace kube-prometheus-stack CRDs on ${cluster}"
        kubectl_do "${cluster}" apply --server-side --force-conflicts -f "${ROOT}"/helmfile.d/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds
      fi

      log_info "  - Upgrade kube-prometheus-stack on ${cluster}"
      helmfile_upgrade "${cluster}" app=prometheus
    done
```

As the `helmfile_upgrade` will do change check then apply this will catch any config drift.
Also please exclude `app=prometheus` in the `80-apply.sh` snippet.